### PR TITLE
Update build instructions for Visual Studio 2019

### LINKED ISSUE #3037: IANA-registered port number for XRP Ledger
After a conversation in #2931, I set out to get an IANA-registered port number for XRP Ledger to user. Working together with @JoelKatz, I'm happy to announce that we now have port **2459**, which shall henceforth be given the official name service name **`xrpl`**:

https://www.iana.org/assignments/service-names-port-numbers/service-names-port-numbers.xhtml?search=2459

So the question is, how should we move forward? Should the new default configuration file migrate from `51235` to `2459`?

The change would complicate things a little bit but I don't expect it to be a breaking change, and nothing stops servers from listening on multiple ports anyways.

### PR DESCRIPTION
**Pull request by drlongle**
## High Level Overview of Change

Update the build instructions for Visual Studio 2019.

Also fix #3037.

### Context of Change

It is helpful to provide build instructions for Visual Studio to make it easier for people to build rippled on Windows. Dependencies are installed using Conan. If you have issues building, it may be helpful to uninstall Visual Studio and use Chocolatey to re-install it. 

### Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactor (non-breaking change that only restructures code)
- [ ] Tests (You added tests for code that already exists, or your new feature included in this PR)
- [x] Documentation Updates
- [ ] Release

### CODE REVIEW

File: cfg/rippled-example.cfg [hunk 1]
```diff
@@ -1617,7 +1617,7 @@ admin = 127.0.0.1
 protocol = http
 
 [port_peer]
-port = 51235
+port = 2459
```
Comment (legleux) [inline cfg/rippled-example.cfg:1620]: I don't think you want to change this.

Comment (intelliot) [reply cfg/rippled-example.cfg:1620]: Apparently, 51235 conflicts with dhcp on Windows. There isn't really a special significance to using 51235 for the peer port. 2459 is the IANA registered port and the [default](https://github.com/XRPLF/rippled/blob/0ce15e0e358996b5371d2adc83e911cc7a3a7225/src/ripple/protocol/SystemParameters.h#L80-L81) used when connecting to other peers (when a port is not specified).

Per https://github.com/XRPLF/rippled/issues/3037, it is probably time to change the port to 2459 in this example config. Operators can always set it to 51235 or whatever they want; this file is just an example.

**Review by intelliot — APPROVED** [review]

File: cfg/rippled-example.cfg (hunk 1 above)
Comment (legleux) [reply cfg/rippled-example.cfg:1620]: Oh I see this is officially the port for xrpl. 
I guess it should have been changed 3 years ago.

File: BUILD.md [hunk 2]
```diff
@@ -170,32 +170,43 @@ Until then, we advise Windows developers to use Visual Studio 2019.
...
 > These instructions assume a basic familiarity with Conan and CMake.
 > If you are unfamiliar with Conan,
 > then please read the [crash course](#a-crash-course-in-cmake-and-conan)
 > at the beginning of this document,
 > or the official [Getting Started][3] walkthrough.
 
+### All Platforms
+
+To build this package, you will need [Python][] (>= 3.7),
+[Conan][] (>= 1.52), and [CMake][] (>= 3.16).
```
Comment (thejohnfreeman) [inline BUILD.md:189]: I do not believe Conan 1.52 is sufficient any more. We should recommend the latest anyway given all the recent changes (1.58).

File: BUILD.md [hunk 3]
```diff
@@ -170,32 +170,43 @@ Until then, we advise Windows developers to use Visual Studio 2019.
...
+If you are unfamiliar with Conan,
+there is a crash course at the end of this document.
```
Comment (thejohnfreeman) [inline BUILD.md:191]: Beginning of the document. Are these changes based off an old version of this document?

File: BUILD.md [hunk 4]
```diff
@@ -170,32 +170,43 @@ Until then, we advise Windows developers to use Visual Studio 2019.
...
+
+```
+conan profile new default --detect
+```
 
 You'll need to compile in the C++20 dialect:
 
 ```
-conan profile update settings.compiler.cppstd=20 default
+conan profile update settings.cppstd=20 default
```
Comment (thejohnfreeman) [inline BUILD.md:206]: This is the wrong command, from an old version of Conan. It was already updated in this document.

File: BUILD.md [hunk 5]
```diff
@@ -170,32 +170,43 @@ Until then, we advise Windows developers to use Visual Studio 2019.
...
+
 [Conan]: https://conan.io/downloads.html
 [CMake]: https://cmake.org/download/
+[Python]: https://www.python.org/downloads/
+
+If you've never used Conan before, use autodetect to setup a default profile:
+
+```
+conan profile new default --detect
+```
```
Comment (thejohnfreeman) [inline BUILD.md:201]: This command is already in the crash course. This section assumes the person has installed and configured Python, Conan, and CMake.

File: BUILD.md [hunk 6]
```diff
@@ -235,40 +254,7 @@ conan profile update env.CC=<path> default
...
-mkdir .build
-cd .build
-conan install .. --output-folder . --build missing --settings build_type=Release --settings compiler.runtime=MT
-conan install .. --output-folder . --build missing --settings build_type=Debug --settings compiler.runtime=MTd
-cmake -DCMAKE_TOOLCHAIN_FILE:FILEPATH=build/generators/conan_toolchain.cmake ..
-cmake --build . --config Release
-cmake --build . --config Debug
-./Release/rippled --unittest
-./Debug/rippled --unittest
-```
```
Comment (thejohnfreeman) [inline BUILD.md:269]: Why are these instructions removed? To what steps do the numbered explanations correspond?

File: Builds/VisualStudio2019/README.md [hunk 7]
```diff
@@ -6,6 +6,12 @@ We do not recommend Windows for rippled production use at this time. Currently,
 the Ubuntu platform has received the highest level of quality assurance,
 testing, and support. Additionally, 32-bit Windows versions are not supported.
 
+While there are several options for building `rippled` on Windows, we recommend using
+[Chocolatey, Conan, CMake and Visual Studio](README.md#build-using-chocolatey-conan-cmake-and-visual-studio-recommended).
+If using this option, you can skip the section [Install Software](README.md#install-software)
+because Conan automatically installs all dependencies for `rippled`.
+
+
```
Comment (thejohnfreeman) [inline Builds/VisualStudio2019/README.md:14]: Editing this document reflects a totally different philosophy on how to present build instructions in this repo. I've got a waiting [PR](https://github.com/XRPLF/rippled/pull/4381) to remove this file.

**Review by thejohnfreeman — CHANGES_REQUESTED** [review]

**Comment by dangell7** [comment]
bump for `.gitignore` :)

**Comment by intelliot** [comment]
FWIW, I would have been fine with merging #4397 as a "quick fix" while this other PR is undergoing review and revisions.

File: BUILD.md [hunk 8]
```diff
@@ -170,32 +170,43 @@ Until then, we advise Windows developers to use Visual Studio 2019.
 
 ## Prerequisites
 
-To build this package, you will need Python (>= 3.7),
-[Conan][] (>= 1.55), and [CMake][] (>= 3.16).
-
 > **Warning**
-> The commands in this document are not meant to be blindly copied and pasted.
+> The commands in this document are guidelines which cannot always be successfully copy and pasted.
```
[the remark this answers is not in the document]
Comment (drlongle) [reply BUILD.md]: Fixed.

File: BUILD.md (hunk 2 above)
Comment (drlongle) [reply BUILD.md]: Fixed.

File: BUILD.md (hunk 4 above)
Comment (drlongle) [reply BUILD.md]: Thanks. Fixed.

File: BUILD.md (hunk 6 above)
Comment (drlongle) [reply BUILD.md]: @justinr1234 has probably removed them by mistake. I put them back.

File: Builds/VisualStudio2019/README.md (hunk 7 above)
Comment (drlongle) [reply Builds/VisualStudio2019/README.md]: Is it better to revert all changes for `Builds/VisualStudio2019/README.md` and add Windows-specific instructions to `BUILD.md`?

File: BUILD.md (hunk 5 above)
Comment (intelliot) [reply BUILD.md:201]: Personally, I think it is good to repeat the command. Obviously, there is a balance (we wouldn't want to repeat it 5 times). But this looks ok to me.

**Comment by intelliot** [comment]
- we can revert the changes to the Windows part
- other changes can be made in BUILD.md

File: BUILD.md (hunk 6 above)
Comment (justinr1234) [reply BUILD.md]: It was removed when there wasn’t a separate readme.

As far as these instructions, it shouldn’t have Debug and Release together like that as both myself and @drlongle ran into issues with mixing debug and release builds back to back.

**Comment by justinr1234** [comment]
@drlongle can you update conflicts and I’ll do a review?

**Comment by thejohnfreeman** [comment]
I'd like to see most of this moved to `docs/environment.md`, like we agreed in https://github.com/XRPLF/rippled/discussions/4433#discussioncomment-5104098.

**Comment by drlongle** [comment]
> I'd like to see most of this moved to `docs/environment.md`, like we agreed in [#4433 (reply in thread)](https://github.com/XRPLF/rippled/discussions/4433#discussioncomment-5104098).

Agreed. Unless there is any objection, let's delete this PR and open a new one.

**Comment by intelliot** [comment]
@drlongle feel free to close this PR and open a new one, at your convenience.

**Pull request closed without merging**